### PR TITLE
Support roles to users

### PR DESCRIPTION
**Pull request by sinnedh**
This PR is based on branch `roles`.

Adding functionality to User
 - GetRoles
 - AssignRole
 - UnassignRoles

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ Following is a list of supported Auth0 resources.
 - [x] [Resource Servers (APIs)](https://auth0.com/docs/api/management/v2#!/Resource_Servers/get_resource_servers)
 - [x] [Roles](https://auth0.com/docs/api/management/v2#!/Roles)
 - [x] [Rules](https://auth0.com/docs/api/management/v2#!/Rules/get_rules)
+- [ ] [Roles](https://auth0.com/docs/api/management/v2#!/Roles/get_roles)
```
Comment (alexkappa) [inline README.md]: Roles is defined just above. We should remove one of them.

File: management/user_test.go [hunk 2]
```diff
@@ -80,11 +92,35 @@ func TestUser(t *testing.T) {
...
 		}
+		t.Logf("%v\n", roles)
+	})
+
+	t.Run("AssignRoles", func(t *testing.T) {
+		roles := []*Role{r1, r2}
+		err = m.User.AssignRoles(auth0.StringValue(u.ID), roles...)
+		if err != nil {
+			t.Error(err)
+			t.Fatal(err)
```
Comment (alexkappa) [inline management/user_test.go]: I think you need just one. Maybe stick to `t.Error`? Not sure why but in most tests I used `Error` unless without the test the following tests wouldn't be useful. Usually when followed by a `DeleteXXX` I avoid fatal errors as they wouldn't run and leave orphaned resources created in the auth0 account.

File: management/user_test.go [hunk 3]
```diff
@@ -132,4 +168,8 @@ func TestUser(t *testing.T) {
 			m.User.Delete(auth0.StringValue(user.ID))
 		}
 	})
+
+	m.Role.Delete(auth0.StringValue(r1.ID))
+	m.Role.Delete(auth0.StringValue(r2.ID))
```
Comment (alexkappa) [inline management/user_test.go]: Perhaps move this close to the creation using `defer`? Just to keep the consistency with other tests.

File: management/user_test.go [hunk 4]
```diff
@@ -28,6 +28,18 @@ func TestUser(t *testing.T) {
 		},
 	}
 
+	r1 := &Role{
+		Name:        auth0.String("admin"),
+		Description: auth0.String("Administrator"),
+	}
+	m.Role.Create(r1)
```
Comment (alexkappa) [inline management/user_test.go]: Good to test for errors here.

**Comment by sinnedh** [comment]
@alexkappa Thanks a lot for your feedback. I have fixed the issues that you have mentioned.

File: management/user_test.go [hunk 5]
```diff
@@ -80,11 +101,34 @@ func TestUser(t *testing.T) {
...
 			t.Error(err)
 		}
+
+	})
+
+	t.Run("UnassignRoles", func(t *testing.T) {
+		roles := []*Role{r1, r2}
+		err = m.User.UnassignRoles(auth0.StringValue(u.ID), roles...)
+		if err != nil {
+			t.Fatal(err)
```
Comment (alexkappa) [inline management/user_test.go]: Can we make this a `t.Error()` please? Then we're good to merge ;)

Comment (sinnedh) [reply management/user_test.go]: Sorry, I misunderstood your previous comment... no its fixed.

Comment (alexkappa) [reply management/user_test.go]: Uhm, did you forget to commit the change? 😄

File: management/job_test.go [hunk 6]
```diff
@@ -1,88 +0,0 @@
-package management
```
Comment (alexkappa) [inline management/job_test.go]: Why did you delete this file?

File: management/user_test.go (hunk 5 above)
Comment (sinnedh) [reply management/user_test.go]: Argh, I have committed the changes to the wrong file (I removed `job_test.go` locally for testing). I have undone these changes now.

File: management/user_test.go [hunk 7]
```diff
@@ -80,13 +101,36 @@ func TestUser(t *testing.T) {
...
+		roles, err = m.User.GetRoles(auth0.StringValue(u.ID))
+		if err != nil {
+			t.Error(err)
+		}
+		t.Logf("%v\n", roles)
+	})
+
+	t.Run("AssignRoles", func(t *testing.T) {
+		roles := []*Role{r1, r2}
+		err = m.User.AssignRoles(auth0.StringValue(u.ID), roles...)
```
Comment (alexkappa) [inline management/user_test.go]: This can also be written using variadic args:

```
m.User.AssignRoles(auth0.StringValue(u.ID), r1, r2)
```

File: management/user_test.go (hunk 5 above)
Comment (alexkappa) [reply management/user_test.go]: Ok. In case you want to run tests in isolation, what I usually do is run a specific test case using the `-run` args to `go test`.

For example running the user tests you could do the following:

```
go test -v ./... -run ^TestUser$
```

File: management/user_test.go (hunk 7 above)
Comment (sinnedh) [reply management/user_test.go]: Nice. Thanks!
